Entities/AHuman.cpp

```diff
@@ -3129,173 +3129,174 @@ void AHuman::Update()
+	float rot = m_Rotation.GetRadAngle();
+
+	// Set Default direction of all the paths!
+	m_Paths[FGROUND][WALK].SetHFlip(m_HFlipped);
+	m_Paths[BGROUND][WALK].SetHFlip(m_HFlipped);
+	m_Paths[FGROUND][CROUCH].SetHFlip(m_HFlipped);
+	m_Paths[BGROUND][CROUCH].SetHFlip(m_HFlipped);
+	m_Paths[FGROUND][CRAWL].SetHFlip(m_HFlipped);
+	m_Paths[BGROUND][CRAWL].SetHFlip(m_HFlipped);
+	m_Paths[FGROUND][ARMCRAWL].SetHFlip(m_HFlipped);
+	m_Paths[BGROUND][ARMCRAWL].SetHFlip(m_HFlipped);
+	m_Paths[FGROUND][CLIMB].SetHFlip(m_HFlipped);
+	m_Paths[BGROUND][CLIMB].SetHFlip(m_HFlipped);
+	m_Paths[FGROUND][STAND].SetHFlip(m_HFlipped);
+	m_Paths[BGROUND][STAND].SetHFlip(m_HFlipped);
+
+	////////////////////////////////////
+	// Jetpack activation and blast direction
+
+	if (m_pJetpack)
+	{
+		if (m_JetTimeTotal > 0) {
+			// Jetpack throttle depletes relative to jet time, but only if throttle range values have been defined
+			float jetTimeRatio = std::max(m_JetTimeLeft / m_JetTimeTotal, 0.0F);
+			m_pJetpack->SetThrottle(jetTimeRatio * 2.0F - 1.0F);
+			float minScale = 1.0F - m_pJetpack->GetMinThrottle();
+			m_pJetpack->SetFlashScale(minScale + (1.0F + m_pJetpack->GetMaxThrottle() - minScale) * jetTimeRatio);
+		}
+		// Start Jetpack burn
+		if (m_Controller.IsState(BODY_JUMPSTART) && m_JetTimeLeft > 0 && m_Status != INACTIVE)
+		{
+			m_pJetpack->TriggerBurst();
+			// This is to make sure se get loose from being stuck
+			m_ForceDeepCheck = true;
+			m_pJetpack->EnableEmission(true);
+			// Quadruple this for the burst
+			m_JetTimeLeft -= g_TimerMan.GetDeltaTimeMS() * 10.0F;
+			if (m_JetTimeLeft < 0) { m_JetTimeLeft = 0; }
+		}
+		// Jetpack is ordered to be burning, or the pie menu is on and was burning before it went on
+		else if ((m_Controller.IsState(BODY_JUMP) || (m_MoveState == JUMP && m_Controller.IsState(PIE_MENU_ACTIVE))) && m_JetTimeLeft > 0)
+		{
+			m_pJetpack->EnableEmission(true);
+			// Jetpacks are noisy!
+			m_pJetpack->AlarmOnEmit(m_Team);
+			// Deduct from the jetpack time
+			m_JetTimeLeft -= g_TimerMan.GetDeltaTimeMS();
+			if (m_JetTimeLeft < 0) { m_JetTimeLeft = 0; }
+			m_MoveState = JUMP;
+			m_Paths[FGROUND][JUMP].Restart();
+			m_Paths[BGROUND][JUMP].Restart();
+		}
+		// Jetpack is off/turning off
+		else
+		{
+			m_pJetpack->EnableEmission(false);
+			if (m_MoveState == JUMP) {
+				m_MoveState = STAND;
+			}
+			// Replenish the jetpack time, twice as fast
+			m_JetTimeLeft += g_TimerMan.GetDeltaTimeMS() * 2;
+			if (m_JetTimeLeft >= m_JetTimeTotal) {
+				m_JetTimeLeft = m_JetTimeTotal;
+			}
+		}
 
-    ////////////////////////////////////
-    // Movement direction
+		// If pie menu is on, keep the angle to what it was before
+		if (m_Controller.IsState(PIE_MENU_ACTIVE))
+		{
+			// Don't change anything
+		}
+		// Direct the jetpack nozzle according to movement stick if analog input is present
+		else if (m_Controller.GetAnalogMove().GetMagnitude() > 0.1F)
+		{
+			//To-do: test whether this works properly
```

Comment:
Have you tested whether this works properly? I.e. do you still need that todo?

Reply:
I'm away from home and don't have a controller with me, lol

Max told me that I flubbed the analog jetpacking at some point so I pushed the fix into this PR.

Reply:
Okay, @MaximDude you're on testing this when you get to this branch I guess.
Gonna leave this comment open for visibility.

Reply:
Note - see issue #322